packages/flutter/lib/src/services/text_input.dart

```diff
@@ -930,4 +915,96 @@ class TextInput {
     }());
     return true;
   }
+
+  MethodChannel _channel;
+
+  TextInputConnection _currentConnection;
+  TextInputConfiguration _currentConfiguration;
+  TextEditingValue _currentTextEditingValue;
+
+  Future<dynamic> _handleTextInputInvocation(MethodCall methodCall) async {
+    if (_currentConnection == null)
+      return;
+    final String method = methodCall.method;
+
+    // Reattach request needs to be handled regardless of the client ID.
+    if (method == 'TextInputClient.requestExistingInputState') {
+      assert(_currentConnection._client != null);
+      _attach(_currentConnection, _currentConfiguration);
```

Comment:
I was wondering here if _currentConfiguration could ever be null at this point. Maybe add an assert(_currentConfiguration != null) above this one, possibly with a comment saying that we always have a configuration if we have a connection?

Reply:
We have an early return at line 926 if `_currentConfiguration == null`.